.github/workflows/ci.yml

```diff
@@ -0,0 +1,218 @@
+name: merge to main workflow
+
+on:
+  push:
+    branches: [main]
+
+env:
+  BRANCH_NAME: ${{ github.event.ref.BRANCH_NAME }}
```

Comment:
I think this just wants to be `guthub.ref_name`, c.f. [here](https://stackoverflow.com/questions/58033366/how-to-get-the-current-branch-within-github-actions)

Reply:
703f74ac7b09d071d28b628257eea0341a07e4ce